src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java

```diff
@@ -356,6 +356,53 @@ public static TlsContextOptions createWithMtlsWindowsCertStorePath(String certif
         return options;
     }
 
+    /**
+     * Helper which creates mutual TLS (mTLS) options using a certificate and private key
+     * stored in a Java keystore.
+     * Will throw an exception if there is no certificate and key at the given certificate alias, or there is some other
+     * error accessing or using the passed-in Java keystore.
+     *
+     * Note: function assumes the passed keystore has already been loaded from a file by calling "keystore.load()" or similar.
+     *
+     * @param keyStore The Java keystore to use. Assumed to be loaded with the desired certificate and key
+     * @param certificateAlias The alias of the certificate and key to use.
+     * @param certificatePassword The password of the certificate and key to use.
+     * @throws CrtRuntimeException if an error occurs, like the keystore cannot be opened or the certificate is not found.
+     * @return A set of options for setting up an mTLS connection
+     */
+    public static TlsContextOptions createWithMtlsJavaKeystore(
+        java.security.KeyStore keyStore, String certificateAlias, String certificatePassword) {
+
+        TlsContextOptions options = new TlsContextOptions();
+        String certificate;
+        try {
+            java.security.cert.Certificate certificateData = keyStore.getCertificate(certificateAlias);
+            if (certificateData == null) {
+                throw new CrtRuntimeException("Certificate at given certificate alias is empty");
+            }
+            String certificateString = StringUtils.simpleBase64Encode(certificateData.getEncoded());
+            certificate = "-----BEGIN CERTIFICATE-----\n" + certificateString + "-----END CERTIFICATE-----\n";
+        } catch (java.security.KeyStoreException | java.security.cert.CertificateEncodingException ex) {
+            throw new CrtRuntimeException("Could not get certificate from Java keystore");
```

Comment:
Seems like the `KeyStoreException` or `CertificateEncodingException` could have useful information in them. Can we just rethrow these wrapped in a RuntimeException?

(same feedback applies everywhere you're catching checked exceptions, and throwing something else)